daphne/src/roles_test.rs

```diff
@@ -367,9 +367,10 @@ async fn http_post_upload_fail_send_invalid_report() {
     );
 }
 
-// This tests a successful Client --> Leader functionality.
+// This currently tests successful Client --> Leader functionality.
```

Comment:
By the way, it is still useful to have this test in addition to a full end-to-end test. It provides more feedback to the developer in case a change is causing one to fail but not the other.

```suggestion
// Test the upload sub-protocol.
```

Reply:
I've added the sub-protocol test as well.